Source/ACE/Entity/Player.cs

```diff
@@ -2343,6 +2355,76 @@ public void HandleActionStackableSplitToContainer(uint stackId, uint containerId
+        /// <param name="containerId">Guid.Full of the container that contains the item</param>
+        /// <param name="amount">Amount taken out of the stack</param>
+        /// <returns></returns>
+        public void HandleActionRemoveItemFromInventory(uint stackId, uint containerId, ushort amount)
+        {
+            ActionChain removeItemsChain = new ActionChain();
+            removeItemsChain.AddAction(this, () =>
+            {
+                Container container;
+                if (containerId == Guid.Full)
+                    container = this;
+                else
+                    container = (Container)GetInventoryItem(new ObjectGuid(containerId));
+
+                if (container == null)
+                {
+                    log.InfoFormat("Asked to remove an item {0} in container {1} - the container was not found",
+                        stackId,
+                        containerId);
+                    return;
+                }
+
+                WorldObject item = container.GetInventoryItem(new ObjectGuid(stackId));
+                if (item == null)
+                {
+                    log.InfoFormat("Asked to remove an item {0} in container {1} - the item was not found",
+                        stackId,
+                        containerId);
+                    return;
+                }
+
+                if (amount >= item.StackSize)
+                    amount = (ushort)item.StackSize;
+
+                ushort oldStackSize = (ushort)item.StackSize;
+                ushort newStackSize = (ushort)(oldStackSize - amount);
+
+                if (newStackSize < 1)
+                {
+                    // Remove item from inventory
+                    DestroyInventoryItem(item);
+                    // Clean up the shard database.
+                    DatabaseManager.Shard.DeleteObject(item.SnapShotOfAceObject(), null);
+                }
+                else
+                {
+                    var valuePerItem = item.Value / item.StackSize;
+                    var burdenPerItem = item.Burden / item.StackSize;
+
+                    item.StackSize = newStackSize;
+                    item.Value = item.StackSize * valuePerItem;
+                    item.Burden = (ushort)(item.StackSize * burdenPerItem);
+
+                    GameMessagePrivateUpdatePropertyInt msgUpdateValue =
+                    new GameMessagePrivateUpdatePropertyInt(container.Sequences, PropertyInt.Value, 1);
+                    Debug.Assert(item.StackSize != null, "stack.StackSize != null");
+                    Debug.Assert(item.Value != null, "stack.Value != null");
+                    GameMessageSetStackSize msgAdjustOldStackSize = new GameMessageSetStackSize(item.Sequences,
+                        item.Guid, (int)item.StackSize, oldStackSize);
+                    CurrentLandblock.EnqueueBroadcast(Location, MaxObjectTrackingRange, msgUpdateValue, msgAdjustOldStackSize);
```

Comment:
Feels wrong to be broadcasting stacksize changes to everyone in range of player, seems like this should only be sent to the player